### PR TITLE
Better error handling

### PR DESCRIPTION
**Pull request by ggggg**
Created a better error handling system for frontend and backend

### CODE REVIEW

File: frontend/pages/courses/[courseID]/admin/create_assignment/index.tsx [hunk 1]
```diff
@@ -1,39 +1,61 @@
...
+                handleError(showSnackSev)
+            );
     };
 
+    const fetchRunners = async () => {
+        const res = await axios.get<RunnerImage[]>("/runner/images").catch(handleError(showSnackSev));
+        if(!res){
+            return;
+        }
+        setRunners(res.data.reduce((a, x) => {
```
Comment (logonoff) [inline frontend/pages/courses/[courseID]/admin/create_assignment/index.tsx:48]: do not use a, x as var names

File: backend/src/common/utils.ts [hunk 2]
```diff
@@ -19,6 +19,12 @@ import {
     onNewTestCase
 } from "./updateScores";
 
+export const errorHandler = (cb: (req: Request, res: Response, next: NextFunction) => Promise<unknown>) => {
```
Comment (logonoff) [inline backend/src/common/utils.ts:22]: use `callback` instead of `cb` for clarity

File: backend/src/common/runner/index.ts [hunk 3]
```diff
@@ -159,6 +169,11 @@ const create_images = () => {
     );
     if(ret?.stdout)
         console.log(ret.stdout.toString());
+    console.log(`Image ${img}/${img_ver} created`);
+};
+const createImages = () => {
+    console.log("creating images");
```
Comment (logonoff) [inline backend/src/common/runner/index.ts:175]: did u forget to remove these

Comment (logonoff) [reply backend/src/common/runner/index.ts:175]: or at least maybe we can separate logging into error, info, warn, etc..?

File: backend/src/common/runner/index.ts [hunk 4]
```diff
@@ -144,10 +151,13 @@ export const queueJob = (job: Job) : Promise<JobResult> => {
     });
 };
 
-const create_images = () => {
-    console.log("creating images");
-    const img = 'python';
-    const img_ver = '3.10.11';
+// Add new images here
+export const images : RunnerImage[] = [
```
Comment (logonoff) [inline backend/src/common/runner/index.ts:155]: feels like this could be a json file

File: backend/src/common/utils.ts [hunk 5]
```diff
@@ -231,17 +237,19 @@ export const deleteFile = async (req: Request, res: Response, table: "solution"
...
+        res.send({message: 'Submission not found.'});
         return;
     }
-    try{
+    try {
         await git.remove({fs, dir: object.git_url, filepath: req.params.file});
         await fs.unlink(`${object!.git_url}/${req.params.file}`);
-    }catch (_) { /* empty */ }
+    } catch (_) {
+        /* empty */
```
Comment (logonoff) [inline backend/src/common/utils.ts:247]: explain why it is empty

**Review by logonoff — APPROVED** [review]
overall looks good

File: backend/src/common/runner/index.ts (hunk 4 above)
Comment (logonoff) [reply backend/src/common/runner/index.ts:155]: or like dynamically generated

File: backend/src/common/runner/index.ts (hunk 3 above)
Comment (ggggg) [reply backend/src/common/runner/index.ts:175]: we use log for all backend logging

File: backend/src/common/utils.ts (hunk 5 above)
Comment (ggggg) [reply backend/src/common/utils.ts:247]: if the file doesnt exist then continue

Comment (ggggg) [reply backend/src/common/utils.ts:247]: comment is there for eslint

File: backend/src/common/runner/index.ts (hunk 4 above)
Comment (ggggg) [reply backend/src/common/runner/index.ts:155]: Probably fine as is for now, we don't really do config files for backend

File: backend/src/common/runner/index.ts (hunk 3 above)
Comment (ggggg) [reply backend/src/common/runner/index.ts]: Fixed on the entire backend now

**Pull request merged**